### PR TITLE
Port to Bootstrap 3

### PR DESCRIPTION
**Pull request by jhass**
This is the continuation for #6002 

I'm not going to work on this personally, instead it will be used to track progress towards merging this to develop. Please see #6002 for the details. To contribute to this please send pull requests against the bootstrap3 branch.

Remaining issues:
- [x] <del>some hounds remarks ([1](https://github.com/diaspora/diaspora/pull/6002#discussion-diff-31260555), [2](https://github.com/diaspora/diaspora/pull/6002#discussion-diff-31343411), [3](https://github.com/diaspora/diaspora/pull/6002#discussion-diff-31343412), [4](https://github.com/diaspora/diaspora/pull/6002#discussion-diff-31343413), [5](https://github.com/diaspora/diaspora/pull/6002#discussion-diff-31350131), [6](https://github.com/diaspora/diaspora/pull/6002#discussion-diff-31350132), [7](https://github.com/diaspora/diaspora/pull/6002#discussion-diff-31379537), [8](https://github.com/diaspora/diaspora/pull/6002#discussion-diff-31379538), [9](https://github.com/diaspora/diaspora/pull/6002#discussion-diff-31379540), [10](https://github.com/diaspora/diaspora/pull/6002#discussion-diff-31379541), [11](https://github.com/diaspora/diaspora/pull/6002#discussion-diff-31379542), [12](https://github.com/diaspora/diaspora/pull/6002#discussion-diff-31381082), [13](https://github.com/diaspora/diaspora/pull/6002#discussion-diff-31381083))</del> #6036
- [x] <del>code refactor ([1](https://github.com/diaspora/diaspora/pull/6002#discussion-diff-31264375), <del>[2](https://github.com/diaspora/diaspora/pull/6002#discussion-diff-31264501)</del>, <del>[3](https://github.com/diaspora/diaspora/pull/6002#discussion-diff-31385096)</del>, <del>[4](https://github.com/diaspora/diaspora/pull/6002#discussion-diff-31385137)</del>)</del> #6024 #6043 #6046
- [x] <del>hash syntax (<del>[1](https://github.com/diaspora/diaspora/pull/6002#discussion-diff-31385150)</del>, <del>[2](https://github.com/diaspora/diaspora/pull/6002#discussion-diff-31385153)</del>, [3](https://github.com/diaspora/diaspora/pull/6002#discussion-diff-31385172))</del> #6037
- [x] <del>inline CSS (<del>[1](https://github.com/diaspora/diaspora/pull/6002#discussion-diff-31385137)</del>, <del>[2](https://github.com/diaspora/diaspora/pull/6002#discussion-diff-31385166)</del>, [3](https://github.com/diaspora/diaspora/pull/6002#discussion-diff-31385167), [4](https://github.com/diaspora/diaspora/pull/6002#discussion-diff-31385179), [5](https://github.com/diaspora/diaspora/pull/6002#discussion-diff-31385183), [6](https://github.com/diaspora/diaspora/pull/6002#discussion-diff-31385186))</del> #6024 #6027 #6050
- [x] <del>[mentions in publisher](https://github.com/diaspora/diaspora/pull/6002#issuecomment-107097619)</del> #6030
- [x] <del>[email icon on password reset page](https://github.com/diaspora/diaspora/pull/6002#issuecomment-107097818)</del> #6032 
- <del>[ ] [hovercard image deformed](https://github.com/diaspora/diaspora/pull/6015#issuecomment-107202867)</del> #6039
- [x] <del>[stream faces misaligned](https://github.com/diaspora/diaspora/pull/6015#issuecomment-107206738)</del> #6017 
- [x] <del>[Getting started page is missing styles](https://github.com/diaspora/diaspora/pull/6015#issuecomment-107253542)</del> #6038
- [x] <del>[Long username breaks notifications dropdown](https://github.com/diaspora/diaspora/pull/6015#issuecomment-107565374)</del> #6049
- [x] <del>[Services icons are misplaced.](https://github.com/diaspora/diaspora/pull/6015#issuecomment-108497874)</del> #6040
- [x] <del>[Fix failing cukes](https://travis-ci.org/diaspora/diaspora/jobs/65299920#L2991)</del> #6042
- [x] <del>[Missing translation for lock account on the admin user search](https://github.com/diaspora/diaspora/pull/6015#issuecomment-108631208)</del>
- [x] <del>[mobile drawer is broken](https://github.com/diaspora/diaspora/pull/6015#issuecomment-108778911)</del>

### CODE REVIEW

**Comment by AugierLe42e** [comment]
Just one more thing I haven't been able to solve: profile picture on hovercards is still deformed. On the other pictures I've added `size: :thumb_small` to `person_image_link` like [here](https://github.com/diaspora/diaspora/pull/6015/files#diff-83f5a1ee6acec6058131736164be0749R8).

hovercard

**Comment by AugierLe42e** [comment]
Meh... Images on stream sidebar are not correctly aligned anymore. Ok, sry, this one is my fault. Submitting a PR right now.

stream

**Comment by jhass** [comment]
Getting started page is missing styles.

**Comment by theworldbright** [comment]
Has the [email icon on password reset page](https://github.com/diaspora/diaspora/pull/6002#issuecomment-107097818) already been fixed?

The email icon looks aligned to me:

screen shot 2015-06-01 at 6 49 50 pm

I'm using Google Chrome Version 43 on the design branch.

**Comment by jhass** [comment]
@theworldbright No, I'm updating the list of issues at the top as they get fixed.

**Comment by theworldbright** [comment]
@jhass Hmm ok. So is this a browser specific issue or am I looking at a different build? The "send me reset instructions" button in the picture you uploaded looks flatter than mine. Which browser was it screenshot-ed in?

**Comment by jhass** [comment]
Firefox, are you sure you're on the right branch

**Comment by jhass** [comment]
Chromium, I think you're looking at the wrong branch :)

**Comment by theworldbright** [comment]
Hmm weird. I switched to the latest bootstrap 3 branch and it still looks the same, even on Firefox. I'll do the other quick fixes first.

**Comment by jhass** [comment]
Try `bin/rake assete:clean tmp:cache:clear`

**Comment by theworldbright** [comment]
I assume you meant `bin/rake assets:clean tmp:cache:clear`

I still see the following in Firefox

screen shot 2015-06-01 at 8 19 21 pm

**Comment by jhass** [comment]
That's weird, it's exactly the old style, here on diasp.org for example https://diasp.org/users/password/new

**Comment by AugierLe42e** [comment]
@theworldbright : Nope, your not on the right branch. The buttons on BS3 are flat. Did you empty your browser cache ?

@jhass : A precision about this icon : it has not the same height than most of others entypo icon, which is why it has alignement problems or sometimes look tinier than the other icons. this is why I don't know how to handle this properly. This icon will have this problem everywhere in the interface (and I suck at CSS :/)

**Comment by AugierLe42e** [comment]
Question : Why not use Backbone on mobile version, too ? To heavy for phones ?

**Comment by theworldbright** [comment]
@AugierLe42e  Clearing the browser catch after doing all that @jhass suggested did the job. Thanks :+1: 

@AugierLe42e  It is interesting how you mention that some desktop views have duplicating mobile views. I was looking at #5076 earlier and I was thinking that an elegant solution would be to fallback to .html.haml when there was no .mobile.haml view (as @Flaburgan has now mentioned). I am interested to see how you refactor the duplicating views.

**Comment by AugierLe42e** [comment]
@theworldbright : you have an exemple [here](https://github.com/diaspora/diaspora/pull/6015/files#diff-62d3acd4937102df71e783387a57a574R1). Both `edit.html.haml` and `edit.mobile.haml` had duplicate code. The trick is only to place the piece of code in an `_edit.haml` partial and make it be rendered in the both views. The big piece of code is just replaced by `= render "partial_name"`.

**Comment by theworldbright** [comment]
Ahh how elegant. I was thinking that I had to mess with the controllers.

**Comment by SansPseudoFix** [comment]
Too long pseudo break design:

bs3_diaspora_too_long_pseudo_notification

Is this solution is ok?

bs3_diaspora_too_long_pseudo_notification_ok

**Comment by Flaburgan** [comment]
@SansPseudoFix god of the overflow is back! I guess an ellipsis would be more elegant.

**Comment by SansPseudoFix** [comment]
> @SansPseudoFix god of the overflow is back!

I can't sleep when I see wrong overflow!

And, my bad, this is the result I wanted:

bs3_diaspora_too_long_pseudo_notification_ok2

**Comment by AugierLe42e** [comment]
> @SansPseudoFix god of the overflow is back!

\o/ .o. \o/

**Comment by AugierLe42e** [comment]
These are pictures from [diaspote.org](http://diaspote.org). There's still some work, but this is way much better than BS2. Especially images are better rendered on mobile version and desktop version is way much easier to read.

screenshot_2015-06-01-23-34-53
screenshot_2015-06-01-23-35-13
screenshot_2015-06-01-23-35-36
screenshot_2015-06-01-23-35-57
screenshot_2015-06-01-23-36-08
screenshot_2015-06-01-23-36-25
screenshot_2015-06-01-23-36-54
screenshot_2015-06-01-23-37-08
screenshot_2015-06-01-23-37-44
screenshot_2015-06-01-23-38-52
screenshot_2015-06-01-23-39-16
screenshot_2015-06-01-23-39-35
screenshot_2015-06-01-23-39-50
screenshot_2015-06-01-23-40-52
screenshot_2015-06-01-23-41-07
screenshot_2015-06-01-23-41-18

**Comment by Flaburgan** [comment]
When did you port the mobile version to bootstrap? With this PR? It was only using blueprint until now.

**Comment by AugierLe42e** [comment]
Nope, it was using BS2. There is no bluepritn anymore in the code. I ported the mobile version at the same time. There was very few work on it, the bigger part was the desktop.

**Comment by AugierLe42e** [comment]
After using that new BS3 interface on diaspote for some hours, I've just noticed a very cool side effect : the interface seems to be much more light and reactive. I thought that heavyness came from Backbone but, actually, even the mobile version a easier to use. I didn't expect BS3 to improve the things that way !

**Comment by theworldbright** [comment]
@jhass I believe inline-css issues [1](https://github.com/diaspora/diaspora/pull/6002#discussion-diff-31385137) has been taken care of by #6024 and [2](https://github.com/diaspora/diaspora/pull/6002#discussion-diff-31385166) by #6027 .

**Comment by jhass** [comment]
Ah, thanks!

**Comment by AugierLe42e** [comment]
@jhass : Hasn't the mentions in publisher been fixed by #6030 ?
Oh, and inline CSS n°2 has been fixed by #6027 ;)

**Comment by jhass** [comment]
Right, my bad.

**Comment by AugierLe42e** [comment]
Eh, BTW, I'v always wondered what this button on mobile view in reaction section was about:

chevron

**Comment by Flaburgan** [comment]
@AugierLe42e it open or close the comments list.

**Comment by AugierLe42e** [comment]
Oh ! I have to admit I never tested ! I'm not really a fan of using icon. In that case, and where it is placed, I don't find this significant. Maybe this should be replaced by a text and placed on the left (like the "reaction" text), don't you think ?

**Comment by jhass** [comment]
> Eh, BTW, I'v always wondered what this button on mobile view in reaction section was about:

Can we keep this out of this PR? Let's keep the scope as narrow as possible.

**Comment by AugierLe42e** [comment]
Sure ;)

**Comment by theworldbright** [comment]
@AugierLe42e Did you fix hash syntax 1 and 2 with #6027 ?

**Comment by theworldbright** [comment]
Does anyone already have ideas as how we will approach the [deformed hovercard issue](https://github.com/diaspora/diaspora/pull/6015#issuecomment-107202867)  ? I was thinking about using [cropit](http://scottcheng.github.io/cropit/) and have the user choose a 400 x 400 cropped square whenever they want to change their hovercard image.

**Comment by jhass** [comment]
In a separate PR maybe, for now please workaround it for example with an overflow.

**Comment by AugierLe42e** [comment]
Notice that, that hovercard problem was already there with BS2 version:

hovercard

(Sorry @Zauberstuhl)

**Comment by jhass** [comment]
Uh, then why did you mention it in the other PR?

**Comment by jhass** [comment]
Services icons are misplaced.

**Comment by AugierLe42e** [comment]
> Uh, then why did you mention it in the other PR?

Because I didn't noticed.

**Comment by jhass** [comment]
Mind opening a separate issue? I'm happy to reduce the scope of this.

**Comment by AugierLe42e** [comment]
#6039 ;)

**Comment by AugierLe42e** [comment]
> Services icons and character count are misplaced.

Where should they be ?

**Comment by jhass** [comment]
left to the cogwheel

**Comment by AugierLe42e** [comment]
On it.

**Comment by Flaburgan** [comment]
Some remarks:
- Looks like the "Welcome, friend" page is higher, so you have to scroll to see "replace this page by a custom...". I didn't have a look at the code, so I don't know if this needs CSS to be modified, in that case it should be done here, or if the grid layout can be improved.
- Doesn't bootstrap 3 provide a way to make the footer sticky? A sticky footer always stays on the bottom of the page regardless of how little content is on the page. However, this footer will be pushed down if there is a lot of content, so it is different from a fixed footer.
- When the stream is empty, "you could search or invite contacts", I thought these were two links, did they disappear, or was this broken before this port?
- On the left sidebar of the stream page, the dropdown to add a followed tag is detached from the input field
- The e-mail link to invite someone is broken (no modal opening), both in the right sidebar and on the search page
- The modal to create an aspect from the dropdown on a profile page is not translated, and the confirmation message too
- The modal to mention someone on his profile page is not displayed (Looks like I have a problem with the modals :P but the one to create an aspect was there)
- On the notifications page, the "mark as read" eye which appears on hover is too close to the right, and too close to the dropdown aspect for the start sharing notifications. That implies that the dropdown aspect sometimes over the window when you have long aspect names.
- The "start a conversation" modal on the contacts page doesn't appear, same result with the "invite someone to join diaspora*" link
- What about the admin panel?

**Comment by jhass** [comment]
> Doesn't bootstrap 3 provide a way to make the footer sticky? A sticky footer always stays on the bottom of the page regardless of how little content is on the page. However, this footer will be pushed down if there is a lot of content, so it is different from a fixed footer.

Out of scope, no porting, discuss in a separate issue.

Some of your other notes sound very familiar, are you sure you tested the latest version of this branch?

**Comment by Flaburgan** [comment]
>Some of your other notes sound very familiar, are you sure you tested the latest version of this branch?

Well, I just git fetched it right before the test, wasn't in my computer before that so I think so.
I found a last remark: the border of the #head div in the SPV is overflowing in the right column.

Thank you everyone for this port, it is a big work! I will definitely pull that on diaspora-fr to have more feedback once this is merged in develop.

**Comment by jhass** [comment]
Uh, `git fetch` does not update your local branch.

**Comment by Flaburgan** [comment]
@jhass I didn't have any local branch, I had to git fetch upstream to then be able to git checkout the branch, so I really think it's up-to-date.

**Comment by jhass** [comment]
> When the stream is empty, "you could search or invite contacts", I thought these were two links, did they disappear, or was this broken before this port?

I don't see any on develop either, please open a separate issue, out of scope.

**Comment by jhass** [comment]
Modals should be fixed with  032d3ae

**Comment by jhass** [comment]
@Flaburgan 
> What about the admin panel?

Do you mean the missing translation for Lock account? 

Or something else?

File: app/assets/templates/aspect_create_modal_tpl.jst.hbs [hunk 1]
```diff
@@ -1,40 +1,51 @@
...
 
-  <div class="modal-header">
-    <button class="close" type="button" data-dismiss="modal" aria-hidden="true">&times;</button>
-    <h3 id="newAspectModalLabel">{{ t "aspects.create.add_a_new_aspect" }}</h3>
-  </div>
+  <div class="modal-dialog">
+    <div class="modal-content">
+      <div class="modal-header">
+        <button class="close" type="button" data-dismiss="modal" aria-hidden="true">&times;</button>
+        <h3 class="modal-title" id="newAspectModalLabel">{{ t "aspects.create.add_a_new_aspect" }}</h3>
```
Comment (jhass) [inline app/assets/templates/aspect_create_modal_tpl.jst.hbs]: @Flaburgan Looks very translated to me?

**Comment by AugierLe42e** [comment]
> Doesn't bootstrap 3 provide a way to make the footer sticky? A sticky footer always stays on the bottom of the page regardless of how little content is on the page. However, this footer will be pushed down if there is a lot of content, so it is different from a fixed footer.

Actually, [it seems like](http://getbootstrap.com/javascript/#affix)

**Comment by jhass** [comment]
Please, let's not clutter this PR with out of scope discussion.

File: app/assets/templates/aspect_create_modal_tpl.jst.hbs (hunk 1 above)
Comment (jhass) [inline app/assets/templates/aspect_create_modal_tpl.jst.hbs]: Actually you're right, it's not translated for a strange reason, but that's already true on develop. Please open a separate issue.

**Comment by AugierLe42e** [comment]
> The e-mail link to invite someone is broken (no modal opening), both in the right sidebar and on the search page

Huh ? How is this possible !? I fixed this ! It worked yesterday !

> The modal to create an aspect from the dropdown on a profile page is not translated, and the confirmation message too

I confirm. Page in french:

trad

> The modal to mention someone on his profile page is not displayed (Looks like I have a problem with the modals :P but the one to create an aspect was there)

I confirm, somebody messed with the modals, I fixed this too. :/

> What about the admin panel?

Didn't check anything but the first page.

**Comment by jhass** [comment]
@AugierLe42e https://github.com/diaspora/diaspora/pull/6015#issuecomment-108630358

**Comment by AugierLe42e** [comment]
Ah, ok, sry for the noise.

**Comment by AugierLe42e** [comment]
Drawer menu on mobile is broken. I've noticed that this morningafter having updated diaspote. I know how to fix this, I'll submit a patch tonight.

drawer

**Comment by Flaburgan** [comment]
This PR reminds me https://github.com/diaspora/diaspora/pull/4410

I like when everybody's working together like that! I should be able to contribute tonight ;)

**Comment by jhass** [comment]
@Flaburgan what's still missing? Steffen just took the liberty to fix the mobile drawer and I was going to clean up the commit history a bit and merge it to develop.

**Comment by AugierLe42e** [comment]
No, emf... @svbergerem removed a modification I made. I removed the `80%` from translate because it was not useful. Drawer doesn't need to translate from 80% of the screen especially on landscape mode. The problem of the drawer is juste because the search bar has not a fixed size lithe menu has. Please let me fix this when I'm home tonight. ;)

**Comment by svbergerem** [comment]
@AugierLe42e The drawer now behaves just like it did before. Let's merge this PR as soon as Travis is happy and you can add your changes on top of this PR afterwards in a new PR since they are not directly related to the BS3 port. Does that sound ok?

**Comment by SansPseudoFix** [comment]
I fixed issue with pagination for conversation in mobile view, but I have some troubles with git (as usual), so... Maybe tonight, maybe tomorrow:

screenshot_2015-06-04-13-35-10

**Comment by SansPseudoFix** [comment]
Done: [#6053](https://github.com/diaspora/diaspora/pull/6053)!

**Comment by jhass** [comment]
Merged via #6055

Good job everybody!

**Comment by AugierLe42e** [comment]
> Does that sound ok?

Emf... Guess I don't have tha choic now :/

**Comment by Flaburgan** [comment]
Congrats to everybody!

**Comment by Flaburgan** [comment]
The work continues on https://github.com/diaspora/diaspora/issues/6063

**Comment by jaywink** [comment]
Awesome stuff everybody! <3

**Comment by goobertron** [comment]
Thanks to everyone who worked on/co-ordinated this effort!

**Pull request closed without merging**